src/app-layer-htp.h

```diff
@@ -76,66 +75,15 @@ enum {
 };
 
 enum {
-    /* libhtp errors/warnings */
-    HTTP_DECODER_EVENT_UNKNOWN_ERROR,
-    HTTP_DECODER_EVENT_GZIP_DECOMPRESSION_FAILED,
-    HTTP_DECODER_EVENT_REQUEST_FIELD_MISSING_COLON,
-    HTTP_DECODER_EVENT_RESPONSE_FIELD_MISSING_COLON,
-    HTTP_DECODER_EVENT_INVALID_REQUEST_CHUNK_LEN,
-    HTTP_DECODER_EVENT_INVALID_RESPONSE_CHUNK_LEN,
-    HTTP_DECODER_EVENT_INVALID_TRANSFER_ENCODING_VALUE_IN_REQUEST,
-    HTTP_DECODER_EVENT_INVALID_TRANSFER_ENCODING_VALUE_IN_RESPONSE,
-    HTTP_DECODER_EVENT_INVALID_CONTENT_LENGTH_FIELD_IN_REQUEST,
-    HTTP_DECODER_EVENT_INVALID_CONTENT_LENGTH_FIELD_IN_RESPONSE,
-    HTTP_DECODER_EVENT_DUPLICATE_CONTENT_LENGTH_FIELD_IN_REQUEST,
-    HTTP_DECODER_EVENT_DUPLICATE_CONTENT_LENGTH_FIELD_IN_RESPONSE,
-    HTTP_DECODER_EVENT_100_CONTINUE_ALREADY_SEEN,
-    HTTP_DECODER_EVENT_UNABLE_TO_MATCH_RESPONSE_TO_REQUEST,
-    HTTP_DECODER_EVENT_INVALID_SERVER_PORT_IN_REQUEST,
-    HTTP_DECODER_EVENT_INVALID_AUTHORITY_PORT,
-    HTTP_DECODER_EVENT_REQUEST_HEADER_INVALID,
-    HTTP_DECODER_EVENT_RESPONSE_HEADER_INVALID,
-    HTTP_DECODER_EVENT_MISSING_HOST_HEADER,
-    HTTP_DECODER_EVENT_HOST_HEADER_AMBIGUOUS,
-    HTTP_DECODER_EVENT_INVALID_REQUEST_FIELD_FOLDING,
-    HTTP_DECODER_EVENT_INVALID_RESPONSE_FIELD_FOLDING,
-    HTTP_DECODER_EVENT_REQUEST_FIELD_TOO_LONG,
-    HTTP_DECODER_EVENT_RESPONSE_FIELD_TOO_LONG,
-    HTTP_DECODER_EVENT_FILE_NAME_TOO_LONG,
-    HTTP_DECODER_EVENT_REQUEST_SERVER_PORT_TCP_PORT_MISMATCH,
-    HTTP_DECODER_EVENT_URI_HOST_INVALID,
-    HTTP_DECODER_EVENT_HEADER_HOST_INVALID,
-    HTTP_DECODER_EVENT_METHOD_DELIM_NON_COMPLIANT,
-    HTTP_DECODER_EVENT_URI_DELIM_NON_COMPLIANT,
-    HTTP_DECODER_EVENT_REQUEST_LINE_LEADING_WHITESPACE,
-    HTTP_DECODER_EVENT_TOO_MANY_ENCODING_LAYERS,
-    HTTP_DECODER_EVENT_ABNORMAL_CE_HEADER,
-    HTTP_DECODER_EVENT_AUTH_UNRECOGNIZED,
-    HTTP_DECODER_EVENT_REQUEST_HEADER_REPETITION,
-    HTTP_DECODER_EVENT_RESPONSE_HEADER_REPETITION,
-    HTTP_DECODER_EVENT_RESPONSE_MULTIPART_BYTERANGES,
-    HTTP_DECODER_EVENT_RESPONSE_ABNORMAL_TRANSFER_ENCODING,
-    HTTP_DECODER_EVENT_RESPONSE_CHUNKED_OLD_PROTO,
-    HTTP_DECODER_EVENT_RESPONSE_INVALID_PROTOCOL,
-    HTTP_DECODER_EVENT_RESPONSE_INVALID_STATUS,
-    HTTP_DECODER_EVENT_REQUEST_LINE_INCOMPLETE,
-    HTTP_DECODER_EVENT_DOUBLE_ENCODED_URI,
-    HTTP_DECODER_EVENT_REQUEST_LINE_INVALID,
-    HTTP_DECODER_EVENT_REQUEST_BODY_UNEXPECTED,
-
-    HTTP_DECODER_EVENT_LZMA_MEMLIMIT_REACHED,
-    HTTP_DECODER_EVENT_COMPRESSION_BOMB,
-
-    HTTP_DECODER_EVENT_RANGE_INVALID,
-
     /* suricata errors/warnings */
-    HTTP_DECODER_EVENT_MULTIPART_GENERIC_ERROR,
-    HTTP_DECODER_EVENT_MULTIPART_NO_FILEDATA,
-    HTTP_DECODER_EVENT_MULTIPART_INVALID_HEADER,
-
-    HTTP_DECODER_EVENT_TOO_MANY_WARNINGS,
-
-    HTTP_DECODER_EVENT_FAILED_PROTOCOL_CHANGE,
+    HTTP_DECODER_EVENT_MULTIPART_GENERIC_ERROR = 200,
```

Comment:
Can this value conflict with another ?

Reply:
If they deserve different integer spaces, maybe they should be even more separated...